### DISCUSSION TITLE
Removal of tinymce configuration - concerned

### DISCUSSION THREAD

**Discussion by AaronSadlerUK** [umbraco/Umbraco-CMS#16242] [Features and ideas]
I'm a bit concerned about this one:
https://github.com/umbraco/Umbraco-CMS/pull/16140

Pretty much every site has some level of custom tinymce required, so removing the configuration without providing any documentation on how to do this using 'client side extension points' is concerning

This would affect the following configuration possibilities:
https://docs.umbraco.com/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/configuration#advanced-configuration

**Comment by iOvergaard** [comment]
Hello Aaron, I concur with your concerns. New TinyMCE documentation will be provided in the V14 documentation, however until then, you can look at some of the built-in Umbraco -> TinyMCE plugins that are conveniently provided through the "tinyMcePlugin" extension point.

[Here is an example of the "View Source Code" button](https://github.com/umbraco/Umbraco.CMS.Backoffice/blob/main/src/packages/tiny-mce/plugins/tiny-mce-code-editor.plugin.ts).

That particular example is registered like this:

```
	{
		type: 'tinyMcePlugin',
		alias: 'Umb.TinyMcePlugin.CodeEditor',
		name: 'Code Editor TinyMCE Plugin',
		js: () => import('./tiny-mce-code-editor.plugin.js'),
		meta: {
			toolbar: [
				{
					alias: 'sourcecode',
					label: 'Source code editor',
					icon: 'sourcecode',
				},
			],
		},
	}
```

**Reply by NikRimington** [reply]
With the move away from "app settings" based configuration - is there a way to have different configs for different instances of Tiny MCE? It was something that AppSettings didn't allow, but it feels like this approach should make it possible?

For example, I would love to be able to have a global config for the styles drop down (as used to be possible with the appSettings approach - but then be able to override it on an instance basis.

Additionally, is the back office based configuration for Tiny MCE Styles going away and/or improving to allow full configuration with all options for the styles? Historically the back office based configuration for styles was very limited, it didn't allow nesting, type targetting, or more advanced style selectors - these limitations are why I didn't use it but instead used the AppSettings config for styles.

**Reply by iOvergaard** [reply]
@NikRimington The new TinyMCE extensions have access to any kind of context when they are spun up. That means they can infer which document type, you are on, which document id, name, language, etc. It's up to you to fill in the blanks and configure the instance as it fits you from then on.

The "Rich Text Editor Stylesheets" will work as before with limited options, however rather than creating a stylesheet, you can now also configure the "styles" option of TinyMCE in a TinyMCE extension. That will allow you to do all sorts of nested menus and whatever else they support.

**Reply by deanleigh** [reply]
So currently I add languages so that users can make a sentence, word or even single character the correct language.
Will that still be possible?
![image](https://github.com/umbraco/Umbraco-CMS/assets/6745929/c1f2d518-9869-4189-acf2-2913a90eb12d)

**Reply by iOvergaard** [reply]
@deanleigh, most definitely. You have access to the tinymce instance and should be able to configure that how you see fit.
